api/groups.go

```diff
@@ -285,3 +285,22 @@ func (api *API) GetGroupHandler(ctx context.Context, w http.ResponseWriter, req
 
 	return models.NewSuccessResponse(jsonResponse, http.StatusOK, nil), nil
 }
+
+//DeleteGroupHandler deletes the group for the given group id
```

Comment:
I would recommend splitting this function up slightly differently whitespace-wise:
```suggestion
func (api *API) DeleteGroupHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
	vars := mux.Vars(req)
	group := models.Group{Name: vars["id"]}
	groupDeleteRequest := group.BuildDeleteGroupRequest(api.UserPoolId)

	_, err := api.CognitoClient.DeleteGroup(groupDeleteRequest)
	if err != nil {
		cognitoErr := models.NewCognitoError(ctx, err, "Cognito DeleteGroup request from Delete group endpoint")
		if cognitoErr.Code == models.NotFoundError {
			return nil, models.NewErrorResponse(http.StatusNotFound, nil, cognitoErr)
		}
		return nil, models.NewErrorResponse(http.StatusInternalServerError, nil, cognitoErr)
	}

	return &models.SuccessResponse{Status: http.StatusOK}, nil
}
```